circuits/src/zk_gadgets/fixed_point.rs

```diff
@@ -129,6 +147,22 @@ impl MultiproverFixedPointGadget {
 
         MultiproverGreaterThanEqZeroGadget::<{DEFAULT_FP_PRECISION + 1}>::constrain_greater_than_eq_zero(diff, fabric, cs)
     }
+
+    /// Constrain a fixed point variable to equal an integral variable when
+    /// floored
+    pub fn constrain_equal_floor(
+        fp: FixedPointVar,
+        integer: Variable,
+        fabric: &Fabric,
+        cs: &mut MpcPlonkCircuit,
+    ) -> Result<(), CircuitError> {
+        let zero_var = cs.zero();
+        let one = ScalarField::one();
+
+        let lhs_minus_rhs =
+            cs.lc(&[fp.repr, integer, zero_var, zero_var], &[one, -*TWO_TO_M_SCALAR, one, one])?;
+        MultiproverGreaterThanEqZeroGadget::<{DEFAULT_FP_PRECISION}>::constrain_greater_than_eq_zero(lhs_minus_rhs, fabric, cs)
```

Comment:
tiny nit: would it be possible to use `MultiproverBitRangeGadget` here? `constrain_greater_than_eq_zero` just calls out to that, but reading this constraint as "the diff must fit in `DEFAULT_FP_PRECISION` bits" makes a lot more sense than "the diff must be >= 0"

Reply:
yeah good call, forgot this gadget existed